system/lib/wasmfs/syscalls.cpp

```diff
@@ -417,6 +420,11 @@ static __wasi_fd_t doOpen(char* pathname,
     return wasmFS.getFileTable().locked().addEntry(openFile);
   }
 
+  // Check user read permissions
+  if (!(parsedPath.child->locked().getMode() & S_IRUSR)) {
```

Comment:
```suggestion
  if (!(parsedPath.child->locked().getMode() & WASMFS_PERM_READ)) {
```

Reply:
Oh wait, actually this is a semantic change. I was previously checking only the user bit, but with this change we check each of the user, group, and other bits, passing the check if any of them allow reading. Is that what we want?

Reply:
I think so - it's what we've done elsewhere. It matches the case where some other user created the file but set the "other" permission, or "group" and we are in that group. (We don't really check group permissions carefully, though. That's been good enough so far in the old FS.)

Reply:
That is, this change matches both the old FS and the new FS code so far.

Reply:
Hmm, we should update our documentation about this then, because that wasn't my understanding after reading it. I guess it's not urgent given that we will eventually replace most of the docs with new WasmFS docs, though.